src/components/Menu/index.vue

```diff
@@ -26,23 +26,29 @@
 </template>
 
 <script lang="ts">
-import { products } from './data'
+import { products } from './data';
 export default {
   name: 'Menu',
   data() {
     return {
       greeting: 'Some Menu' as string,
       menuProducts: products,
-      filterButtons: '' as string
-    }
+      filterButtons: '' as string,
+      selectedCategory: 'all',
+    };
   },
   methods: {
-    removeDuplicateCategory() {
-      console.log('clicked btn')
-    }
+    removeDuplicateCategory(): void {
+      console.log('clicked btn');
+    },
   },
-  computed: {}
-}
+  computed: {
+    menuProductsCategory(): string[] {
+      const Categories = this.menuProducts.map((item) => item.category);
+      return ['all', ...new Set(Categories)];
```

Comment:
Change Categories to lowercase, we are mapping over products, so I suggest importing Product type and changing (item) => item.category to (product: Product) => product.category